File: src/sorting/bubble_sort.py
def bubble_sort(nums: list[int]) -> list[int]:
    swapping = True
    while swapping:
        swapping = False
        for i in range(1, len(nums)):
            if nums[i] < nums[i-1]:
                nums[i-1], nums[i] = nums[i], nums[i-1]
                swapping = True
    return nums


if __name__ == "__main__":
    tests = [
            [1, 8, 4, 3, 9, 10, 0]
            ]
    for t in tests:
        print(bubble_sort(t))

File: src/sorting/BubbleSort.java
import java.util.Arrays;
public class BubbleSort {
    public static void main(String[] args) {
        int[] unsortedArray = new int[args.length];
        for (int i = 0; i < args.length; i++) {
            unsortedArray[i] = Integer.parseInt(args[i]);
        }

        Boolean swapping = true;
        while (swapping) {
            swapping = false;
            for (int i = 1; i < unsortedArray.length; i++) {
                if (unsortedArray[i] < unsortedArray[i - 1]) {
                    swapping = true;
                    int tmp = unsortedArray[i];
                    unsortedArray[i] = unsortedArray[i - 1];
                    unsortedArray[i - 1] = tmp;
                }
            }
        }
        IO.println(Arrays.toString(unsortedArray));
    }
}


